### DISCUSSION TITLE
Sanity Check and Tensorboard Stats

### DISCUSSION THREAD

**Discussion by AffeZwei** [acids-ircam/RAVE#100] [Show and tell]
Hi there, first of all thanks for all the work that you have done with this great project. I did a Windows build for nn~ and it works great with the voices .ts file. Now I am trying to make my own .ts files on a set of lupophone samples. I have only about 56 minutes of samples for the data set, and have completed just over 1Mill steps and am pretty happy with the results, even though I realize the data set is relatively small. Since I am really only a user-end user at this point, I would like to share the progress so far, so maybe someone could let me know if this looks ok, or I should make changes. I am running on collab, ca. 2-3 hours a day, and have clocked over 25 hours so far. (Can't get CuDNN to work with on either of my laptops Windows or Linux.)  
Here is a link to the soundfile from step 1058694 https://drive.google.com/file/d/1nabcD_2qTFHgQvat7XBi3GmEenn5qVJi/view?usp=sharing

Here are some stats from the Tensorboard. Any thoughts? I know after 1M steps there should be an increase in the distance. Question 1) I know what pred_true is, but what do the numbers mean? I am puzzled that there are below zero numbers.
Question 2) Is there a good way to know when to stop training? Should I just keep listening or keep my eyes on the distance/validation values? (probably both?) Or will the program know best and stop for me?

![lupo3_pred_true](https://user-images.githubusercontent.com/62151108/179076266-dcdafcb2-dba3-44c0-89b3-3e24888dbdb3.png)
![lupo3_validation](https://user-images.githubusercontent.com/62151108/179076271-62433eab-1cc7-42c9-ad17-56e4493974c3.png)
![lupo3_distance](https://user-images.githubusercontent.com/62151108/179076273-f4b2eb89-541a-4b84-a19c-3a82ed799f71.png)

**Comment by lang216** [comment]
May I know why "after 1M steps there should be an increase in the distance"?

**Reply by lang216** [reply]
Actually I think the number going up is because the learning rate is too big. I solved this problem by lowering the learning rate.

**Reply by AffeZwei** [reply]
In my layman's understanding, the first part (warmup) is the representational learning, where the distance and the validation values should draw closer together. However, the second part (after 1M or so steps), is the adversarial fine tuning, for which the distance and validation are not as relevant. More relevant are the predictions true and false. I am curious though, did lowering the learning rate during the second part improve the acoustic results?
